### DISCUSSION TITLE
Error: Could not find locale while pre-rendering page, make sure you called `setStaticParamsLocale` at the top of your pages

### DISCUSSION THREAD

**Discussion by aryankarim** [vercel/next.js#65060] [Help]
### Summary

I get this error when building for prod:
```javascript
Error: Could not find locale while pre-rendering page, make sure you called `setStaticParamsLocale` at the top of your pages
```
plus this at the end
```javascirpt
> Export encountered errors on following paths:
        /_not-found/page: /_not-found
        /page: /
```

### Additional information

```js
Using 
`    "next-international": "^1.2.4",`
with
`    "next": "14.2.1",`
```

**Comment by turjmann** [comment]
any info on this yet?

**Comment by danielivanovz** [comment]
i am encountering the same error

**Comment by aryankarim** [comment]
@turjmann & @danielivanovz can you guys provide how to reproduce this error -- a Stackblitz link or something? tbh I forgot how i fixed this.

**Reply by danielivanovz** [reply]
personally got it fixed - i had the following structure

```
src
├── [locale]
│   ├── @first
│   │   ├── layout.tsx
│   │   └── ...
│   ├── @second
│   │   ├── layout.tsx
│   │   └── ...
│   └── layout.tsx
│       
├── layout.tsx
├── not-found.tsx
└── global-error.tsx
```

moved to:

```
src
└── [locale]
    ├── @first
    │   ├── layout.tsx
    │   └── ...
    ├── @second
    │   ├── layout.tsx
    │   └── ...
    │── layout.tsx
    └── not-found.tsx
    └── global-error.tsx
```

tried to replicate here as well: https://stackblitz.com/edit/stackblitz-starters-x4vzz7?file=package.json

**Reply by aryankarim** [reply]
I've used `[locale]/[...rest]/page.tsx` to catch all to not found pages. If your file structure solution works, that is cleaner.

**Comment by ErwannRousseau** [comment]
I use `next-international` and had this error too, so I added this to my main `layout.tsx`:
```ts
import { getStaticParams } from '@/locales/server
 
export function generateStaticParams() {
  return getStaticParams()
}
```

The build error was resolved, but I caught a 500 error in the preview deployment.

In my build logs I had this:
```bash
○  (Static)   prerendered as static content
●  (SSG)      prerendered as static HTML (uses getStaticProps)
ƒ  (Dynamic)  server-rendered on demand
```

So I've removed the `generateStaticParams()` function from the `layout.tsx` and it now works for me

**Reply by igokul1973** [reply]
You added it and immediately removed? Very unclear explanation.
